### PR TITLE
Add support for yarn

### PR DESCRIPTION
**Pull request by eddiewebb**
additional boolean paramter to use yarninstead of npm ci.
that condition is used for install and cache steps through all jobs and
commands.

This is offered as potential solution to #65 an #28

### CODE REVIEW

File: src/orb.yml [hunk 1]
```diff
@@ -65,22 +65,45 @@ commands:
...
           keys:
             - cache-{{ arch }}-{{ .Branch }}-{{ checksum "package.json" }}
       # it would be nice to vary this command based on presence of "package-lock.json"
       # and also allow using yarn or pnpm
-      - run:
-          name: 'Npm CI'
-          command: 'npm ci'
+      - when:
+          condition: << parameters.yarn >>
+          steps:
```
Comment (bahmutov) [inline src/orb.yml]: ha, I did not know one could add steps into `when` condition like that, this is pretty slick

File: src/orb.yml [hunk 2]
```diff
@@ -65,22 +65,45 @@ commands:
...
       - run: npx cypress verify
       # save new cache folder if needed
-      - save_cache:
-          key: cache-{{ arch }}-{{ .Branch }}-{{ checksum "package.json" }}
-          paths:
-            - ~/.npm
-            - ~/.cache
+      - when:
+          condition: << parameters.yarn >>
+          steps:
```
Comment (bahmutov) [inline src/orb.yml]: and especially that the steps could be `run` and `save_cache` and any other command

**Comment by bahmutov** [comment]
Perfect, @eddiewebb I will merge this, and will add an example and will be happy in general, thank you so much.

**Comment by tymondesigns** [comment]
Thank you @eddiewebb!

File: src/orb.yml (hunk 1 above)
Comment (eddiewebb) [reply src/orb.yml]: Super powerful.  Also can use empty strings will evaluate to false.

**Pull request merged**